dashboard/app/controllers/plc/user_course_enrollments_controller.rb

```diff
@@ -34,17 +34,32 @@ def create
     enrolled_users, nonexistent_users, nonteacher_users, other_failure_users, other_failure_reasons =
       Plc::UserCourseEnrollment.enroll_users(user_emails, user_course_enrollment_params[:plc_course_id])
 
-    notice_string = enrolled_users.empty? ? '' : "Enrollments created for #{listify(enrolled_users)}<br/>"
-    notice_string += "The following users did not exist #{listify(nonexistent_users)}<br/>" unless nonexistent_users.empty?
-    notice_string += "The following users were not teachers #{listify(nonteacher_users)}<br/>" unless nonteacher_users.empty?
-    notice_string += "The following users failed for other reasons #{listify(other_failure_users)}<br/>" unless other_failure_users.empty?
-    notice_string += "The failures were because of reasons #{listify(other_failure_reasons)}" unless other_failure_reasons.empty?
+    notice_string = enrolled_users.empty? ? '' : "#{enrolled_users.length} enrollment(s) created#{listify_first_ten(enrolled_users)}<br/>"
+    notice_string += "#{nonexistent_users.length} user(s) did not exist#{listify_first_ten(nonexistent_users)}<br/>" unless nonexistent_users.empty?
+    notice_string += "#{nonteacher_users.length} user(s) were not teachers#{listify_first_ten(nonteacher_users)}<br/>" unless nonteacher_users.empty?
+    notice_string += "#{other_failure_users.length} user(s) failed for other reasons#{listify_first_ten(other_failure_users)}<br/>" unless other_failure_users.empty?
```

Comment:
I didn't realize we concatenated all of these messages together and was a little worried that even with limits, this notice is potentially too long... but counted and the Honeybadger error report was a case with more than 117 email addresses in the list.  It looks like our new maximum is 40, plus a little fixed-length copy, so I think we're safe.

Reply:
yeah I thought about that too--the limit is pretty large so I think we should be safe. My main concern was if the "other" error messages were very long we could run into issues, but I think that's unlikely since we have only seen the too long error once.